### DISCUSSION TITLE
add new parameter to update_with_detections method

### DISCUSSION THREAD

**Discussion by Ali-Fayzi** [roboflow/supervision#851] [Q&A]
### Search before asking

- [X] I have searched the Supervision [issues](https://github.com/roboflow/supervision/issues) and found no similar feature requests.

### Description

To create and add new tracking methods, please add the 'frame' parameter to the 'update_with_detections' function.
`def update_with_detections(detections, frame):`

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by onuralpszr** [comment]
Hello, @Ali-Fayzi  could you elaborate bit more why should we add this ? Also could you explain how is it effect you as well ? Because as for tracking go this is how we using it. 

```python
import supervision as sv
from ultralytics import YOLO
import numpy as np
from supervision.assets import download_assets, VideoAssets
import cv2

model = YOLO("yolov8l.pt")
byte_tracker = sv.ByteTrack()
download_assets(VideoAssets.VEHICLES)

start_point = sv.Point(869, 1041)
end_point = sv.Point(2900, 1041)

bounding_box_annotator = sv.BoundingBoxAnnotator()
label_annotator = sv.LabelAnnotator()
line_zone_annotator = sv.LineZoneAnnotator(
    thickness=2,
    color=sv.Color.WHITE,
    text_thickness=2,
    text_color=sv.Color.BLACK,
    text_scale=2.5,
    text_offset=1.5,
    text_padding=10,
    display_in_count=False,
    display_out_count=True,
)
trace_annotator = sv.TraceAnnotator()
line_zone = sv.LineZone(start_point, end_point)

def callback(frame: np.ndarray, index: int) -> np.ndarray:
    results = model(frame)[0]
    detections = sv.Detections.from_ultralytics(results)
    detections = byte_tracker.update_with_detections(detections)

    labels = [f"#{tracker_id}" for tracker_id in detections.tracker_id]

    annotated_frame = bounding_box_annotator.annotate(
        scene=frame.copy(), detections=detections
    )
    annotated_frame = label_annotator.annotate(
        scene=annotated_frame, detections=detections, labels=labels
    )
    annotated_frame = trace_annotator.annotate(
        scene=annotated_frame, detections=detections
    )

    line_zone.trigger(detections)
    annotated_frame = line_zone_annotator.annotate(
        frame=annotated_frame, line_counter=line_zone
    )

    return annotated_frame

sv.process_video(
    source_path="vehicles.mp4", target_path="result.mp4", callback=callback
)


```

As you can see we have 

```python
detections = byte_tracker.update_with_detections(detections)
```

We do update with tracker information and send as detections and return detection and use the frame we use and annotate it so I did not see why do we need too add frame ? Since we already have frame and what problem do we solve in here ?

**Comment by Ali-Fayzi** [comment]
Hello @onuralpszr , thank you for your response. I am going to work on adding new tracking methods to Super Vision. For instance, in the Strong Sort , feature extraction from the frame is required. Therefore, by adding a second parameter `frame=None` to the `update_with_detections(detections,frame=None)` function, developers can extend new tracking methods smoothly without any issues.

**Comment by onuralpszr** [comment]
> Hello @onuralpszr , thank you for your response. I am going to work on adding new tracking methods to Super Vision. For instance, in the Strong Sort , feature extraction from the frame is required. Therefore, by adding a second parameter `frame=None` to the `update_with_detections(detections,frame=None)` function, developers can extend new tracking methods smoothly without any issues.

I understand and first of "update_with_detections" is function to belong the byterack and there is a another one in smoother but they both "seperated" function and there is no generic function in for "update_with_detections" so when you adding strong-short you can do differently and before I write my answer I checked why do you need frame for first you needed 2 things I notice and first is for extract h,w and secondly use cases of `Shi-Tomasi Corner Detector` but If you gonna implement strongsort couple points I can say, %100 you need to figure out how make it library free because we really really don't want to add "pytorch" as dependency or big tracker library like "boxmot" I noticed in your previous PR  you were using it. So I am closing this issue because technically it is your end to handle I hope you will succeed as well. Good luck

**Comment by onuralpszr** [comment]
I am also converting to discussion

**Comment by Ali-Fayzi** [comment]
In my new implementation, I removed the dependency on the "boxmot" library. You mentioned that you don't want to add pytorch as a dependency, but in practice, when you need to install the "ultralytics" library, you actually require the installation of pytorch as well.

Source: https://github.com/ultralytics/ultralytics?tab=readme-ov-file#documentation

Supervision+strongsort_tracker Notebook: https://colab.research.google.com/drive/1K1r4e2jYC8G6CgB2gD040n25XhLU0JG5
